trappy/base.py

```diff
@@ -265,6 +265,24 @@ def generate_parsed_data(self):
 
             yield data_dict
 
+    def optimize_dataframe(self):
+        """Optimize memory footprint by setting minimal data types required by
+           each column"""
+        for col in self.data_frame.columns:
+            if self.data_frame[col].dtype.kind == 'i':
+                self.data_frame[col].apply(pd.to_numeric, downcast='signed')
+                continue
+            if self.data_frame[col].dtype.kind == 'f':
+                self.data_frame[col].apply(pd.to_numeric, downcast='float')
+                continue
+            # Convert string objects (pointer) to categories, only when we have
+            # a relatively limited number of unique values (50% of the rows)
+            num_unique_values = len(self.data_frame[col].unique())
```

Comment:
Interesting. I would have done `len(set(self.data_fram[col]))`, but I guess if the dataframe has a `.unique()` method it has to be faster than `set()`.